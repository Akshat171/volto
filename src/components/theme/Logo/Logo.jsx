/**
 * Logo component.
 * @module components/theme/Logo/Logo
 */

import { defineMessages, useIntl } from 'react-intl';
import { Image } from 'semantic-ui-react';
import { useSelector } from 'react-redux';
import config from '@plone/volto/registry';
import { UniversalLink } from '@plone/volto/components';
import LogoImage from '@plone/volto/components/theme/Logo/Logo.svg';
import { URLUtils } from '@plone/volto/helpers';

const messages = defineMessages({
  site: {
    id: 'Site',
    defaultMessage: 'Site',
  },
  plonesite: {
    id: 'Plone Site',
    defaultMessage: 'Plone Site',
  },
});

/**
 * Logo component class.
 * @function Logo
 * @param {Object} intl Intl object
 * @returns {string} Markup of the component.
 */
const Logo = () => {
  const { isMultilingual } = config.settings;
  const lang = useSelector((state) => state.intl.locale);
  const intl = useIntl();
  const path = URLUtils.normalizePath(isMultilingual ? `/${lang}` : '');
  return (
    <UniversalLink href={path} title={intl.formatMessage(messages.site)}>
      <Image
        src={LogoImage}
        alt={intl.formatMessage(messages.plonesite)}
        title={intl.formatMessage(messages.plonesite)}
      />
    </UniversalLink>
  );
};

export default Logo;
