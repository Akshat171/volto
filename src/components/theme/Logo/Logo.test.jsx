import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import Logo from './Logo';

const mockStore = configureStore();

describe('Logo', () => {
  it('renders a logo component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
      router: {
        location: {
          pathname: '/',
        },
      },
      site: {
        data: {},
      },
      navroot: {
        data: {
          url: '/',
          title: 'My site',
        },
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Logo />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
