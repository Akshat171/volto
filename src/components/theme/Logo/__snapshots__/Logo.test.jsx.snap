// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders a logo component 1`] = `
<a
  className={null}
  href="/"
  onClick={[Function]}
  target={null}
  title="My site"
>
  <img
    alt="My site"
    className="ui image"
    src="Logo.svg"
    title="My site"
  />
</a>
`;
